libsolidity/interface/FileReader.cpp

```diff
@@ -153,7 +153,7 @@ ReadCallback::Result FileReader::readFile(string const& _kind, string const& _so
 			return ReadCallback::Result{false, "File outside of allowed directories."};
 
 		if (!boost::filesystem::is_regular_file(candidates[0]))
-			return ReadCallback::Result{false, "Not a valid file."};
+			return ReadCallback::Result{false, "Not a valid existing file: " + candidates[0].string()};
```

Comment:
The file does exist, there's a `boost::filesystem::exists()` check above. At this point it can just be something other than a regular file (directory, device, broken link, etc.)

```suggestion
			return ReadCallback::Result{false, "Not a regular file: " + candidates[0].string()};
```